### PR TITLE
datastreams: add subjects

### PR DESCRIPTION
**Pull request by wgresshoff**
:heart: Thank you for your contribution!

### Description

Adds the subject datastreams to match https://github.com/inveniosoftware/invenio-vocabularies/pull/355.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by wgresshoff** [comment]
This PR has a dependency on https://github.com/inveniosoftware/invenio-vocabularies/pull/355 which has to be merged first.

File: invenio_app_rdm/config.py [hunk 1]
```diff
@@ -123,6 +124,12 @@
 from invenio_vocabularies.contrib.names.datastreams import (
     VOCABULARIES_DATASTREAM_WRITERS as NAMES_WRITERS,
 )
+from invenio_vocabularies.contrib.subjects.datastreams import (
```
Comment (ptamarit) [inline invenio_app_rdm/config.py:127]: [Subjects also has readers defined](https://github.com/inveniosoftware/invenio-vocabularies/blob/master/invenio_vocabularies/contrib/subjects/datastreams.py#L33-L34), so I think they should be added here too.

**Review by ptamarit — COMMENTED** [review]
The tests should get green after a rebase.

**Comment by wgresshoff** [comment]
This is replaced by https://github.com/inveniosoftware/invenio-app-rdm/pull/2790

**Pull request closed without merging**